test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/ObjectJSTest.scala

```diff
@@ -30,4 +32,13 @@ class ObjectJSTest {
     (new js.Object: Any).asInstanceOf[Object]
     (js.Array(5)  : Any).asInstanceOf[Object]
   }
+
+  @Test def cloneOnNonScalaObject(): Unit = {
+    class CloneOnNonScalaObject extends js.Object {
+      def boom(): Any = this.clone()
+    }
+
+    val obj = new CloneOnNonScalaObject()
+    assertThrows(classOf[CloneNotSupportedException], obj.boom())
```

Comment:
`(new js.Object).clone()` instead? Or did you want to test this for a non-native JS class specifically?

Reply:
It's `protected` ;)

Reply:
Ah...